crates/relayer/src/link/cli.rs

```diff
@@ -92,25 +93,23 @@ impl<ChainA: ChainHandle, ChainB: ChainHandle> Link<ChainA, ChainB> {
         )
         .entered();
 
-        let (sequences, src_response_height) = if maybe_sequences.is_empty() {
-            // Find the sequence numbers of unreceived packets
-            unreceived_packets(
-                self.a_to_b.dst_chain(),
-                self.a_to_b.src_chain(),
-                &self.a_to_b.path_id,
-            )
-            .map_err(LinkError::supervisor)?
-        } else {
-            (
-                maybe_sequences,
-                self.a_to_b.src_chain().query_latest_height().unwrap(),
-            )
-        };
+        // Find the sequence numbers of unreceived packets
+        let (mut sequences, src_response_height) = unreceived_packets(
```

Comment:
@ancazamfir I assumed that we want to allow half-open ranges (eg. `2000..` to mean all packets with sequence number 2000 or higher) or even just large ranges, we cannot enumerate all of them as there may be up to 2^64 of them. Therefore we have to query the unreceived packets/acks and then filter by the ranges we are given.

The downside is that we now do have to query for those even if we know exactly which packets we want to clear. The upside is that we won't attempt to query for packets which have already been relayed.

Does that trade-off work for you?

Reply:
I think it should be fine. We can think of further optimizations in the future if needed.